client/components/site-selector/index.jsx

```diff
@@ -393,6 +393,7 @@ const SiteSelector = React.createClass( {
 				<Search
 					ref="siteSearch"
 					onSearch={ this.onSearch }
+					delaySearch={ true }
```

Comment:
I'd rather not, as it's just another path to test. We could drop the default debounce delay if it feels too slow? Felt fine to me though.